components/playlist/playlist_types.h

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PLAYLIST_PLAYLIST_TYPES_H_
+#define BRAVE_COMPONENTS_PLAYLIST_PLAYLIST_TYPES_H_
+
+#include <string>
+#include <vector>
+
+namespace playlist {
+
+struct PlaylistChangeParams {
+  enum class ChangeType {
+    kChangeTypeNone,
+    kChangeTypeAdded,            // New playlist added but not ready state
+    kChangeTypeThumbnailReady,   // Thumbnail ready to use for playlist
+    kChangeTypeThumbnailFailed,  // Failed to fetch thumbnail
+    kChangeTypePlayReady,        // Playlist ready to play
+    kChangeTypeDeleted,          // A playlist deleted
+    kChangeTypeAllDeleted,       // All playlist are deleted
+    kChangeTypeAborted,          // Aborted during the creation process
+  };
+  static std::string GetPlaylistChangeTypeAsString(
+      PlaylistChangeParams::ChangeType type);
+
+  PlaylistChangeParams();
+  PlaylistChangeParams(ChangeType type, const std::string& id);
+  ~PlaylistChangeParams();
+
+  ChangeType change_type = ChangeType::kChangeTypeNone;
+  std::string playlist_id;
+};
+
+struct MediaFileInfo {
+  MediaFileInfo(const std::string& url, const std::string& title);
+  ~MediaFileInfo();
+
+  std::string media_file_url;
+  std::string media_file_title;
+};
+
+struct CreatePlaylistParams {
+  CreatePlaylistParams();
+  CreatePlaylistParams(const CreatePlaylistParams& rhs);
+  CreatePlaylistParams& operator=(const CreatePlaylistParams& rhs);
+  CreatePlaylistParams(CreatePlaylistParams&& rhs);
+  CreatePlaylistParams& operator=(CreatePlaylistParams&& rhs);
+  ~CreatePlaylistParams();
+
+  std::string playlist_thumbnail_url;
+  std::string playlist_name;
+  std::vector<MediaFileInfo> video_media_files;
+  std::vector<MediaFileInfo> audio_media_files;
+};
+
+// TODO(simonhong): Rename to PlaylistItemInfo
+struct PlaylistInfo {
+  PlaylistInfo();
+  PlaylistInfo(const PlaylistInfo& rhs);
+  PlaylistInfo& operator=(const PlaylistInfo& rhs);
+  PlaylistInfo(PlaylistInfo&& rhs);
+  PlaylistInfo& operator=(PlaylistInfo&& rhs);
```

Comment:
sorry for being too picky, but move constructors/operators should be marked `noexcept`. even though we have exceptions disabled, this is still required.

https://google.github.io/styleguide/cppguide.html#noexcept
> Specifying move constructors as noexcept improves performance in some cases, e.g., std::vector<T>::resize() moves rather than copies the objects if T's move constructor is noexcept

Reply:
> sorry for being too picky, but move constructors/operators should be marked noexcept. even though we have exceptions disabled, this is still required.

No worries. I didn't know that there's performance improvement with `noexcept`. Thanks!